{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8b86f52c",
   "metadata": {},
   "source": [
    "### **Atividade Complementar de Bioinformática**\n",
    "#### André Nakano Miyake - Caso 2\n",
    "\n",
    "Comecei configurando o ambiente, instalando todas as bibliotecas necessárias num ambiente WSL Ubuntu. \n",
    "> Descrição: Paciente lactente bebê, sexo masculino, 6 meses, apresentada na consulta de pediatria geneticista devido a triagem neonatal alterada evidenciando níveis elevados de fenilalanina e atraso incipiente no desenvolvimento neuropsicomotor.\n",
    "\n",
    "> Procedimento: Painel genético específico para análise de variantes no gene PAH, realizado a partir de amostra de sangue.\n",
    "\n",
    "> Objetivo: Identificar variantes genéticas responsáveis pela fenilcetonúria (PKU), fornecendo base para diagnóstico precoce e orientações terapêuticas individualizadas para prevenção de complicações neurológicas.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09311f3c",
   "metadata": {},
   "source": [
    "#### 1. Controle de qualidade"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb25d719",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Gerando o relatório da qualidade dos reads de sequenciamento\n",
    "fastqc -o ../fastqc/ ../raw/caso2-sindrome-metabolica_R1.fastq.gz ../raw/caso2-sindrome-metabolica_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69cfd449",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Filtrar sequências muito pequenas (-m 90) e cortar as pontas\n",
    "cutadapt \\\n",
    "  -u 5 -U 5 \\\n",
    "  -u -5 -U -5 \\\n",
    "  -m 90 \\\n",
    "  -o ../cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz \\\n",
    "  -p ../cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz \\\n",
    "  ../raw/caso2-sindrome-metabolica_R1.fastq.gz \\\n",
    "  ../raw/caso2-sindrome-metabolica_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf326adb",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Gerando o relatório da qualidade dos reads de sequenciamento, novamente, após o corte\n",
    "fastqc -o ../fastqc/ ../cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz ../cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1487fec",
   "metadata": {},
   "source": [
    "Ao analisar os HTMLs, notei que a qualidade ainda estava ruim, mas decidi prosseguir com os arquivos que tenho para seguir com o fluxo da atividade. A primeira imagem da esquerda é o original sem cutadapt, e a segunda da direita é a após o corte do cutadapt.\n",
    "\n",
    "<img src=\"download.png\" width=\"500\" alt=\"Not trimmed\"> <img src=\"download-1.png\" width=\"500\" alt=\"Trimmed\"> \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4fc3c0e",
   "metadata": {},
   "source": [
    "#### 2. Mapeamento com a referência do genoma humano"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "2a1c7efc",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 1.18 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=266550618, availableWord=30755168\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 50732074 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 93723498 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 131930362 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 165884698 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 196059354 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 222874650 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 246704106 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 266550618 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 80 iterations.\n",
      "[bwa_index] 69.95 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.62 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.72 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 36.20 sec\n",
      "[main] Version: 0.7.19-r1273\n",
      "[main] CMD: bwa index ../reference/Chr12-reference.fasta\n",
      "[main] Real time: 119.530 sec; CPU: 108.664 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# Indexando o genoma de referência (Chr12) para o BWA e Samtools\n",
    "samtools faidx ../reference/Chr12-reference.fasta\n",
    "bwa index ../reference/Chr12-reference.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c948bd83",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 71422 sequences (10000189 bp)...\n",
      "[M::process] read 71440 sequences (10000217 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34793, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.01, 28.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71422 reads in 17.108 CPU sec, 17.321 real sec\n",
      "[M::process] read 71420 sequences (10000023 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
      "[M::mem_pestat] mean and std.dev: (340.12, 28.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71440 reads in 17.449 CPU sec, 16.824 real sec\n",
      "[M::process] read 71434 sequences (10000161 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.72, 28.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71420 reads in 19.011 CPU sec, 18.446 real sec\n",
      "[M::process] read 71458 sequences (10000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.06, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71434 reads in 20.328 CPU sec, 20.567 real sec\n",
      "[M::process] read 71410 sequences (10000066 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34737, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.92, 28.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71458 reads in 21.098 CPU sec, 20.518 real sec\n",
      "[M::process] read 71432 sequences (10000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34709, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.95, 28.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71410 reads in 19.803 CPU sec, 20.061 real sec\n",
      "[M::process] read 71406 sequences (10000180 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34758, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.12, 28.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71432 reads in 18.565 CPU sec, 17.988 real sec\n",
      "[M::process] read 71428 sequences (10000008 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34743, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.14, 28.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71406 reads in 19.452 CPU sec, 18.848 real sec\n",
      "[M::process] read 71424 sequences (10000062 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34792, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.76, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71428 reads in 17.905 CPU sec, 18.101 real sec\n",
      "[M::process] read 71404 sequences (10000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34710, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.81, 28.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71424 reads in 18.681 CPU sec, 18.136 real sec\n",
      "[M::process] read 71426 sequences (10000119 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34719, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.01, 28.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71404 reads in 18.050 CPU sec, 18.262 real sec\n",
      "[M::process] read 71442 sequences (10000251 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34721, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.92, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 18.475 CPU sec, 17.914 real sec\n",
      "[M::process] read 71406 sequences (10000003 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34766, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
      "[M::mem_pestat] mean and std.dev: (339.93, 28.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71442 reads in 21.868 CPU sec, 21.267 real sec\n",
      "[M::process] read 71442 sequences (10000199 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34732, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.28, 28.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71406 reads in 20.067 CPU sec, 20.361 real sec\n",
      "[M::process] read 71412 sequences (10000159 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34768, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.90, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71442 reads in 18.384 CPU sec, 17.778 real sec\n",
      "[M::process] read 71408 sequences (10000266 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34721, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.47, 28.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71412 reads in 18.502 CPU sec, 18.730 real sec\n",
      "[M::process] read 71436 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34759, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.75, 28.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71408 reads in 18.546 CPU sec, 17.984 real sec\n",
      "[M::process] read 71416 sequences (10000223 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.89, 28.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71436 reads in 20.213 CPU sec, 19.623 real sec\n",
      "[M::process] read 71428 sequences (10000242 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34719, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.92, 28.24)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71416 reads in 18.078 CPU sec, 18.287 real sec\n",
      "[M::process] read 71454 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.06, 28.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71428 reads in 14.910 CPU sec, 14.255 real sec\n",
      "[M::process] read 71418 sequences (10000260 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.01, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71454 reads in 18.625 CPU sec, 18.878 real sec\n",
      "[M::process] read 71430 sequences (10000091 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34701, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.03, 28.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71418 reads in 19.865 CPU sec, 19.307 real sec\n",
      "[M::process] read 71430 sequences (10000055 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34748, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.66, 28.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71430 reads in 15.941 CPU sec, 16.139 real sec\n",
      "[M::process] read 71438 sequences (10000091 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34743, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.97, 28.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71430 reads in 19.964 CPU sec, 19.394 real sec\n",
      "[M::process] read 71416 sequences (10000236 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34711, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.99, 28.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71438 reads in 16.149 CPU sec, 15.532 real sec\n",
      "[M::process] read 71422 sequences (10000179 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34778, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.66, 28.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71416 reads in 15.033 CPU sec, 15.191 real sec\n",
      "[M::process] read 71468 sequences (10000090 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34724, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.85, 28.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71422 reads in 14.587 CPU sec, 14.752 real sec\n",
      "[M::process] read 71438 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34735, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.72, 28.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71468 reads in 14.682 CPU sec, 14.054 real sec\n",
      "[M::process] read 71434 sequences (10000169 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34776, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.87, 28.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71438 reads in 15.026 CPU sec, 15.212 real sec\n",
      "[M::process] read 71426 sequences (10000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.06, 28.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71434 reads in 15.045 CPU sec, 14.334 real sec\n",
      "[M::process] read 71444 sequences (10000231 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34750, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.18, 28.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 16.466 CPU sec, 16.709 real sec\n",
      "[M::process] read 71424 sequences (10000085 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34727, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.06, 28.41)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71444 reads in 16.093 CPU sec, 15.337 real sec\n",
      "[M::process] read 71400 sequences (10000006 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34686, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.96, 28.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71424 reads in 15.378 CPU sec, 15.618 real sec\n",
      "[M::process] read 71420 sequences (10000057 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34701, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.91, 28.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71400 reads in 14.873 CPU sec, 14.147 real sec\n",
      "[M::process] read 71462 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34731, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (339.98, 28.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71420 reads in 15.298 CPU sec, 15.429 real sec\n",
      "[M::process] read 71448 sequences (10000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.00, 28.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71462 reads in 14.932 CPU sec, 14.289 real sec\n",
      "[M::process] read 71440 sequences (10000263 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34758, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.04, 28.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71448 reads in 15.682 CPU sec, 15.857 real sec\n",
      "[M::process] read 71406 sequences (10000050 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34679, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.89, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71440 reads in 16.018 CPU sec, 15.392 real sec\n",
      "[M::process] read 71438 sequences (10000263 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34691, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.83, 28.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71406 reads in 16.915 CPU sec, 17.085 real sec\n",
      "[M::process] read 71426 sequences (10000071 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34702, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.91, 28.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71438 reads in 17.450 CPU sec, 16.881 real sec\n",
      "[M::process] read 71434 sequences (10000215 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34781, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.88, 28.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 15.069 CPU sec, 15.226 real sec\n",
      "[M::process] read 71454 sequences (10000170 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34750, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.55, 28.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71434 reads in 15.668 CPU sec, 15.052 real sec\n",
      "[M::process] read 71414 sequences (10000053 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34731, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.08, 28.37)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71454 reads in 16.169 CPU sec, 16.274 real sec\n",
      "[M::process] read 71436 sequences (10000101 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34692, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.95, 28.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71414 reads in 15.484 CPU sec, 14.830 real sec\n",
      "[M::process] read 71428 sequences (10000197 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34725, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.96, 28.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71436 reads in 16.033 CPU sec, 16.190 real sec\n",
      "[M::process] read 71430 sequences (10000071 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.81, 28.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71428 reads in 16.244 CPU sec, 15.673 real sec\n",
      "[M::process] read 71426 sequences (10000066 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (240, 440)\n",
      "[M::mem_pestat] mean and std.dev: (340.24, 28.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (200, 480)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71430 reads in 16.233 CPU sec, 16.393 real sec\n",
      "[M::process] read 71438 sequences (10000081 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.90, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 16.218 CPU sec, 15.566 real sec\n",
      "[M::process] read 71426 sequences (10000055 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34762, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.02, 28.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71438 reads in 14.478 CPU sec, 14.653 real sec\n",
      "[M::process] read 71430 sequences (10000131 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34733, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.04, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 14.512 CPU sec, 13.842 real sec\n",
      "[M::process] read 71426 sequences (10000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34769, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.98, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71430 reads in 14.421 CPU sec, 14.629 real sec\n",
      "[M::process] read 71430 sequences (10000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34745, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.78, 28.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71426 reads in 15.314 CPU sec, 14.671 real sec\n",
      "[M::process] read 71400 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34757, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (339.99, 28.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71430 reads in 14.046 CPU sec, 14.187 real sec\n",
      "[M::process] read 71464 sequences (10000254 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34700, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.78, 28.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71400 reads in 13.946 CPU sec, 13.305 real sec\n",
      "[M::process] read 71396 sequences (10000284 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34736, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.86, 28.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71464 reads in 13.937 CPU sec, 14.091 real sec\n",
      "[M::process] read 71454 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34686, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.06, 28.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71396 reads in 14.332 CPU sec, 14.478 real sec\n",
      "[M::process] read 71436 sequences (10000230 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34709, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 360)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (237, 442)\n",
      "[M::mem_pestat] mean and std.dev: (340.04, 28.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (196, 483)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71454 reads in 14.193 CPU sec, 13.540 real sec\n",
      "[M::process] read 71418 sequences (10000140 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34707, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.90, 28.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71436 reads in 13.926 CPU sec, 14.072 real sec\n",
      "[M::process] read 71400 sequences (10000124 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34742, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.69, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71418 reads in 14.121 CPU sec, 13.472 real sec\n",
      "[M::process] read 71420 sequences (10000002 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34711, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.78, 28.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71400 reads in 14.092 CPU sec, 14.240 real sec\n",
      "[M::process] read 71402 sequences (10000259 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34774, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.87, 28.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71420 reads in 14.426 CPU sec, 13.791 real sec\n",
      "[M::process] read 71410 sequences (10000193 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34754, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (320, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (242, 437)\n",
      "[M::mem_pestat] mean and std.dev: (340.13, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (203, 476)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71402 reads in 17.152 CPU sec, 17.372 real sec\n",
      "[M::process] read 71460 sequences (10000088 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34740, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.85, 28.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71410 reads in 16.484 CPU sec, 15.843 real sec\n",
      "[M::process] read 71418 sequences (10000274 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34733, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.99, 28.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71460 reads in 14.244 CPU sec, 14.397 real sec\n",
      "[M::process] read 71464 sequences (10000278 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34708, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.87, 28.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71418 reads in 14.950 CPU sec, 14.333 real sec\n",
      "[M::process] read 71416 sequences (10000161 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34779, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.95, 28.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71464 reads in 14.001 CPU sec, 14.141 real sec\n",
      "[M::process] read 71476 sequences (10000115 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34785, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.85, 28.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71416 reads in 13.913 CPU sec, 14.050 real sec\n",
      "[M::process] read 71434 sequences (10000196 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34720, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.62, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71476 reads in 14.409 CPU sec, 13.790 real sec\n",
      "[M::process] read 71398 sequences (10000300 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34695, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.61, 28.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71434 reads in 14.189 CPU sec, 14.351 real sec\n",
      "[M::process] read 71420 sequences (10000253 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34694, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.09, 28.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71398 reads in 13.996 CPU sec, 13.268 real sec\n",
      "[M::process] read 71390 sequences (10000051 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34703, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.74, 28.35)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71420 reads in 14.176 CPU sec, 14.406 real sec\n",
      "[M::process] read 71422 sequences (10000184 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34739, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (340.15, 28.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71390 reads in 13.935 CPU sec, 13.189 real sec\n",
      "[M::process] read 71412 sequences (10000289 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34768, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.96, 28.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71422 reads in 13.779 CPU sec, 13.932 real sec\n",
      "[M::process] read 71402 sequences (10000196 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34738, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.75, 28.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71412 reads in 13.981 CPU sec, 13.403 real sec\n",
      "[M::process] read 71442 sequences (10000137 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34756, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.67, 28.23)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71402 reads in 14.079 CPU sec, 14.227 real sec\n",
      "[M::process] read 71444 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34722, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.89, 28.45)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71442 reads in 14.079 CPU sec, 14.229 real sec\n",
      "[M::process] read 71448 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34781, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.88, 28.43)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71444 reads in 13.927 CPU sec, 13.281 real sec\n",
      "[M::process] read 71444 sequences (10000284 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34725, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.75, 28.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71448 reads in 13.935 CPU sec, 14.090 real sec\n",
      "[M::process] read 71440 sequences (10000195 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34716, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.71, 28.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71444 reads in 13.999 CPU sec, 13.350 real sec\n",
      "[M::process] read 71416 sequences (10000256 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34749, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.68, 28.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71440 reads in 14.110 CPU sec, 14.262 real sec\n",
      "[M::process] read 71444 sequences (10000220 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34741, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.86, 28.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71416 reads in 14.067 CPU sec, 13.419 real sec\n",
      "[M::process] read 71456 sequences (10000138 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34746, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.97, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71444 reads in 14.200 CPU sec, 14.349 real sec\n",
      "[M::process] read 71402 sequences (9994959 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34739, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.87, 28.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71456 reads in 13.914 CPU sec, 13.266 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 34786, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (319, 339, 359)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (239, 439)\n",
      "[M::mem_pestat] mean and std.dev: (339.77, 28.28)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (199, 479)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 71402 reads in 13.829 CPU sec, 14.139 real sec\n",
      "[main] Version: 0.7.19-r1273\n",
      "[main] CMD: bwa mem -R @RG\\tID:caso2\\tSM:caso2\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1 ../reference/Chr12-reference.fasta ../cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz ../cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz\n",
      "[main] Real time: 1325.925 sec; CPU: 1343.338 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# O 'bwa mem' lê os arquivos .gz e a referência, e 'samtools sort' recebe a saída de arquivo sam e cria o arquivo .bam\n",
    "bwa mem ../reference/Chr12-reference.fasta \\\n",
    "  -R '@RG\\tID:caso2\\tSM:caso2\\tPL:ILLUMINA\\tLB:lib1\\tPU:unit1' \\\n",
    "  ../cutadapt/caso2-sindrome-metabolica-trimmed_R1.fastq.gz \\\n",
    "  ../cutadapt/caso2-sindrome-metabolica-trimmed_R2.fastq.gz \\\n",
    "  | samtools sort -o ../bwa/caso2-sindrome-metabolica-mapped-host.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03cbf978",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# Cria um arquivo .bai na mesma pasta do BAM (../bwa/)\n",
    "samtools index ../bwa/caso2-sindrome-metabolica-mapped-host.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8689a809",
   "metadata": {},
   "source": [
    "Abrindo o arquivos no IGV..."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d29a2e8c",
   "metadata": {},
   "source": [
    "#### 3. Chamada de variantes"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bioinfo",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
